modules/flyte/groups.tf

```diff
@@ -1,14 +1,22 @@
-resource "azuread_group" "flyte_metadata_access" {
-  display_name     = "Flyte metadata access"
+resource "azuread_group" "flyte_metadata_group" {
```

Comment:
I would group these definitions, for example, 
```
locals {
  flyte_azuread_group_members = {
    metadata = [
      azurerm_user_assigned_identity.this["flyte_controlplane"].principal_id,
      azurerm_user_assigned_identity.this["flyte_dataplane"].principal_id
    ]
    data = [azurerm_user_assigned_identity.this["flyte_dataplane"].principal_id]
    sas = [azurerm_user_assigned_identity.this["flyte_dataplane"].principal_id]
  }
}

resource "azuread_group" "flyte" {
  for_each         = local.flyte_azuread_group_members
  display_name     = "${var.deploy_id}-flyte-${each.key}"
  security_enabled = true
  members          = each.value
}

```

also keeping the naming consistent with other resources

Reply:
@miguelhar Thank you for taking the time to illustrate with a code example.  I'm not yet a big fan of grouping resource definitions in a dict, mostly for readability reasons (as noted in the comment below).  Let's leave the definitions as-is for now and we can revisit this topic later.

Your recommendation on consistent naming convention does offer some inspiration for cleaning up.  Added a commit to:
- Prefix all resources with the deploy id
- Drop resource type suffix in TF local names (since they are always referenced along with the resource type anyway)
- Add resource type suffix to Azure resource names where appropriate, e.g. `${var.deploy_id}-flyte-metadata-group`